test/e2e/network/proxy.go

```diff
@@ -258,9 +262,102 @@ var _ = SIGDescribe("Proxy", func() {
 				framework.Failf(strings.Join(errs, "\n"))
 			}
 		})
+
+		ginkgo.It("proxy connection returns a series of 301 redirections for a pod", func() {
+			ns := f.Namespace.Name
+			httpVerbs := []string{"DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"}
+			var statusCode int
+
+			ginkgo.By("Creating a Pod")
+			_, err := f.ClientSet.CoreV1().Pods(ns).Create(context.TODO(), &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "proxy-pod-target",
+					Labels: map[string]string{
+						"test": "redirect"},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Image: imageutils.GetE2EImage(imageutils.Agnhost),
+						Name:  "agnhost",
+					}},
+					RestartPolicy: v1.RestartPolicyNever,
+				}}, metav1.CreateOptions{})
+			framework.ExpectNoError(err, "failed to create pod")
+			framework.Logf("pod created")
+
+			err = wait.PollImmediate(podRetryPeriod, podRetryTimeout, checkPodStatus(f, "test=redirect"))
+			framework.ExpectNoError(err, "Pod didn't start within time out period")
+
+			transportCfg, err := f.ClientConfig().TransportConfig()
+			framework.ExpectNoError(err, "Error accessing TransportConfig")
+			tlsCfg, err := transport.TLSConfigFor(transportCfg)
+			framework.ExpectNoError(err, "Error accessing TLSConfigFor(transportCfg)")
+			// Disable 301 automatic follow
+			client := &http.Client{
+				CheckRedirect: func(req *http.Request, via []*http.Request) error {
+					return http.ErrUseLastResponse
+				},
+				Transport: &http.Transport{TLSClientConfig: tlsCfg},
+			}
+
+			for _, httpVerb := range httpVerbs {
+
+				// Using RESTClient doesn't work due to auto follow on 301
+				framework.Logf("Using http verb: %v", httpVerb)
+				r := f.ClientSet.CoreV1().RESTClient().Verb(httpVerb).
+					Namespace(ns).
+					Resource("pods").
+					SubResource("proxy").
+					Name("proxy-pod-target")
+				framework.Logf("Request String: %v", r.URL().String())
+				// Request returns 500 status - Can this be diabled?
+				_, err := r.Do(context.Background()).StatusCode(&statusCode).Raw()
+				framework.Logf("RESTClient() StatusCode: %d", statusCode)
```

Comment:
just a question: Do we expect 500 as a status code here?

Reply:
I was hoping to use the normal `RESTClient` to check that each request was being redirected. As the `RESTClient` auto follows the redirection it looks like using the `http.Client` is the only option to catch the 301 status code.

I'll remove that section of the test as it was more for reference/discussion point on why `http.Client` was needed.

Reply:
I think that is a good idea. The removal will be able to make the test more clear and simple.

Reply:
Removed